File: 백준/Silver/1966. 프린터 큐/프린터 큐.cc
#include <iostream>
#include <queue>

using namespace std;

int main() {
    int T;
    cin >> T;

    for (int i = 0; i < T; i++)
    {
        int range, key;
        cin >> range >> key; 
        queue<pair<int,int>> q;
        priority_queue<int> pq;
        int cnt = 0;

        for (int j = 0; j < range; j++)
        {
            int value; 
            cin >> value;
            q.push({j, value}); // 인덱스, 중요도
            pq.push(value);
        }

        int curr = 0;
        while (!q.empty()) {
            pair<int, int> temp = q.front();
            q.pop();

            if (temp.second < pq.top()) {
                q.push(temp);
            } else if (temp.second == pq.top()) {
                cnt++;
                pq.pop();
                if (temp.first == key) {
                    break;
                }
            }
        }
        
        cout << cnt << endl;
    }
    
}

File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
from queue import PriorityQueue, Queue

T = input()
for _ in range(int(T)):
  ran, key = map(int, input().split())
  q = Queue()
  pq = PriorityQueue()
  cnt = 0

  values = list(map(int, input().split()))
  for j in range(ran):
    value = values[j]
    q.put({'key': j, 'value': value})
    pq.put(-value)
  
  while not q.empty():
    temp = q.get()
    top = pq.get()

    if temp['value'] < -top:
      q.put(temp)
      pq.put(top)
    else:
      cnt += 1
      if temp['key'] == key: break

  print(cnt)